File: Algorithms/String_Algorithms/Palindrome.java
/**
 * Compile and execute: java -classpath Palindrome.jar Palindrome.java
 *
 * @author  Ytalo Ramon
 */

package String_Algorithms;

public class Palindrome {

    public static void main(String[] args) {

        String inputs[] = { "ana", "carl", "anaa", "level", "levell", "a", "deified", "deiffied", "civicc", "tesett",
                "vuc", "amaamaamaama" };

        for (String value : inputs)
            System.out.println("\"" + value + "\" is palindrome? " + (is_palindrome(value) ? "Yes" : "No"));

    }

    static boolean is_palindrome(String value) {
        int i = -1, j = value.length();
        while (i < j && value.charAt(++i) == value.charAt(--j));

        return j <= i;
    }
}

File: Algorithms/String_Algorithms/Palindrome.c
/** 
 * @author  Ytalo Ramon
 */

#include "stdio.h"
#include "stdlib.h"
#include "string.h"

#define STRINGSIZE 11

int is_palindrome(char *word);

int main(int argc, char const *argv[]){
    
    char inputs[STRINGSIZE][STRINGSIZE] = {
        "ana",
        "carl",
        "anaa",
        "level",
        "levell",
        "a",
        "deified",
        "deiffied",
        "civicc",
        "tesett",
        "vuc"
    };

    for(int i = 0; i < STRINGSIZE; i++)
        printf("\"%s\" is palindrome ? %s\n", inputs[i],
                is_palindrome(inputs[i]) ? "Yes" : "NO");
    
    return 0;
}

int is_palindrome(char *word){
    int i = -1, j = strlen(word);
    while (i < j && word[++i] == word[--j]);
    
    return j <= i;
}